### PR TITLE
Site store getters renaming

### PR DESCRIPTION
**Pull request by maxme**
`getSelfHostedSite` gradually became another kind of method to get sites that are accessed via XMLRPC, These sites could be Jetpack sites (installed or connected) or pure self hosted sites depending how the user addded the site in the app.

To make these getters less confusing, we renamed these them to getSitesAccessedViaXMLRPC / getSitesAccessedViaWPComRest - they now reflects the data model more accurately.

I also removed `getJetpackSites` and his siblings, it's not used in the app and it's misleading: what's a Jetpack site? a self hosted with Jetpack installed? or Connected? What about Jetpack sites accessed via XMLRPC?

Note: I kept `getLocalIdForSelfHostedSiteIdAndXmlRpcUrl` because it's used in a very specific case (migration) in wpandroid, but once we drop migration code, we can drop that getter too.

### CODE REVIEW

File: fluxc/src/main/java/org/wordpress/android/fluxc/store/SiteStore.java [hunk 1]
```diff
@@ -403,15 +403,23 @@ public boolean hasSiteWithLocalId(int id) {
     /**
      * Returns all .COM and Jetpack sites in the store.
```
Comment (aforcier) [inline fluxc/src/main/java/org/wordpress/android/fluxc/store/SiteStore.java]: We should probably update these comments to be more specific about 'Jetpack'.

File: fluxc/src/main/java/org/wordpress/android/fluxc/store/SiteStore.java [hunk 2]
```diff
@@ -403,15 +403,23 @@ public boolean hasSiteWithLocalId(int id) {
...
      * Returns the number of .COM and Jetpack sites in the store.
      */
-    public int getWPComAndJetpackSitesCount() {
-        return SiteSqlUtils.getWPComAndJetpackSites().getAsCursor().getCount();
+    public int getSitesAccessedViaWPComRestCount() {
+        return SiteSqlUtils.getSitesAccessedViaWPComRest().getAsCursor().getCount();
+    }
+
+    /**
+     * Checks whether the store contains at least one site accessed via XMLRPC (self-hosted sites or
```
Comment (aforcier) [inline fluxc/src/main/java/org/wordpress/android/fluxc/store/SiteStore.java]: Copy/paste? 😛

File: fluxc/src/main/java/org/wordpress/android/fluxc/store/SiteStore.java [hunk 3]
```diff
@@ -447,60 +456,23 @@ public boolean hasWPComSite() {
     /**
      * Returns all self-hosted sites (can't be Jetpack) in the store.
```
Comment (aforcier) [inline fluxc/src/main/java/org/wordpress/android/fluxc/store/SiteStore.java]: Here too, 'can't be Jetpack' is unclear now.

**Comment by aforcier** [comment]
What about [this call](https://github.com/wordpress-mobile/WordPress-Android/blob/ea576cc07c9a039984a3aa3cc9f21b863b824600/WordPress/src/main/java/org/wordpress/android/util/AnalyticsUtils.java#L62) to `hasJetpackSite()`?

**Comment by maxme** [comment]
> What about this call to hasJetpackSite()?

We'll replace that one by `getSitesAccessedViaWPComRestCount() - getWPComSitesCount() > 0` and comment it in `AnalyticsUtils`.

**Comment by maxme** [comment]
Updated getter comments, ready for another pass.

**Comment by aforcier** [comment]
:shipit:!

**Pull request merged**